pkg/datapath/linux/node_ids.go

```diff
@@ -92,6 +92,12 @@ func (n *linuxNodeHandler) allocateIDForNode(node *nodeTypes.Node) (uint16, erro
 		}
 	}
 
+	// If we failed to allocate nodeID, don't map any IP to 0 nodeID.
+	// This causes later errors like "Found a foreign IP address with the ID of the current node"
+	if nodeID == uint16(idpool.NoID) {
+		return nodeID, errs
+	}
```

Comment:
Can't we simply return on line 86? I think I just forgot the return statement there.